test/TrueLayer.AcceptanceTests/MockBank/MockBankClient.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Mime;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+
+namespace TrueLayer.AcceptanceTests.MockBank;
+
+public class MockBankClient
+{
+    private readonly HttpClient _httpClient;
+
+    public MockBankClient(HttpClient httpClient)
+    {
+        _httpClient = httpClient;
+    }
+
+    public async Task<Uri> AuthorisePaymentAsync(
```

Comment:
I think we do something hacky already for similar purposes, for [payments](https://github.com/TrueLayer/truelayer-dotnet/blob/99d7cc1750ef873380b1b2f12409804be5ebd2c1/test/TrueLayer.AcceptanceTests/PaymentTests.cs#L669) and [mandates](https://github.com/TrueLayer/truelayer-dotnet/blob/99d7cc1750ef873380b1b2f12409804be5ebd2c1/test/TrueLayer.AcceptanceTests/MandatesTests.cs#L337)

Could we make this function usable for every resource type (mandates and payments) and reuse this everywhere?
Not needed now, we can do this in another PR

Reply:
yeah, I was planning to do a refactor later, we can optimise several part of the tests.
The problem with this [method](https://github.com/TrueLayer/truelayer-dotnet/blob/99d7cc1750ef873380b1b2f12409804be5ebd2c1/test/TrueLayer.AcceptanceTests/PaymentTests.cs#L669) was that returns a GetPaymentResponse.PaymentDetails and we need the specific type of the GetPaymentResponse for this test.